build_for_ultimaker.sh

```diff
@@ -43,13 +43,93 @@ DEB_DIR=`pwd`/debian
 KCONFIG=`pwd`/configs/${BUILDCONFIG}_config
 KERNEL_BUILD=`pwd`/_build_armhf/${BUILDCONFIG}-linux
 
+INITRAMFS_COMPRESSION="${INITRAMFS_COMPRESSION:-.lzo}"
+INITRAMFS_ROOT_GID=${INITRAMFS_ROOT_GID:-0}
+INITRAMFS_ROOT_UID=${INITRAMFS_ROOT_UID:-0}
+INITRAMFS_SOURCE="initramfs/initramfs.list"
+INITRAMFS_IMG="${KERNEL_BUILD}/initramfs.cpio${INITRAMFS_COMPRESSION}"
+
+GEN_INIT_CPIO="${KERNEL_BUILD}/usr/gen_init_cpio"
+GEN_INITRAMFS_LIST="${KERNEL}/scripts/gen_initramfs_list.sh"
+
+BB_PKG="http://ftp.nl.debian.org/debian/pool/main/b/busybox/busybox-static_1.22.0-19+b3_armhf.deb"
+BB_BIN="busybox"
+
 # Set the release version if it's not passed to the script
 RELEASE_VERSION=${RELEASE_VERSION:-9999.99.99}
 
 # Initialize repositories
 git submodule init
 git submodule update
 
+busybox_get()
+{
+    local START_DIR=$(pwd)
+    local BB_DIR=$(mktemp -d)
+    local BB_AR="${BB_DIR}/busybox-static_armhf.deb"
+    local INITRAMFS_DIR="${1}"
+
+    if [ ! -d "${INITRAMFS_DIR}" ]; then
+        echo "No initramfs dir set to download busybox into."
+        exit 1
+    fi
+
+    if [ -z "${BB_DIR}" ]; then
+        echo "Unable to create temporary directory to get busybox."
+        exit 1
+    fi
+
+    cd "${BB_DIR}"
+    wget -q "${BB_PKG}" -O "${BB_AR}"
+    ar -x "${BB_AR}" "data.tar.xz"
+    tar -xf "${BB_DIR}/data.tar.xz" --strip=2 "./bin/${BB_BIN}"
+    cd "${START_DIR}"
+    mv "${BB_DIR}/${BB_BIN}" "${INITRAMFS_DIR}"
+    rm -r "${BB_DIR}"
+    if [ ! -x "${INITRAMFS_DIR}/${BB_BIN}" ]; then
+        echo "Failed to get busybox."
+        exit 1
+    fi
+}
+
+initramfs_prepare()
+{
+    local INITRAMFS_SRC="$(pwd)/initramfs"
+    local INITRAMFS_DST="${KERNEL_BUILD}/initramfs"
+
+    if [ ! -d "${INITRAMFS_DST}" ]; then
+        mkdir -p "${INITRAMFS_DST}"
+    fi
+
+    cp -a "${INITRAMFS_SRC}/" "${INITRAMFS_DST}/../"
+
+    if [ ! -x "${INITRAMFS_DST}/${BB_BIN}" ]; then
+        busybox_get "${INITRAMFS_DST}"
+    fi
+}
+
+initramfs_build()
+{
+    if [ ! -x "${GEN_INIT_CPIO}" ]; then
+        kernel_build
+    else
+        initramfs_prepare
+    fi
+    if [ ! -x "${GEN_INIT_CPIO}" ]; then
+        echo "Kernel failed to create gen_init_cpio."
+        exit 1
+    fi
+
+    START_DIR=$(pwd)
+    cd "${KERNEL_BUILD}"
+    ${GEN_INITRAMFS_LIST} \
+        -o "${INITRAMFS_IMG}" \
+        -u "${INITRAMFS_ROOT_UID}" \
+        -g "${INITRAMFS_ROOT_GID}" \
+        "${START_DIR}/${INITRAMFS_SOURCE}"
```

Comment:
we use the SOURCE list file here, that is wrong as we need to use the in line 104 copied version